js/date-picker/js/lumx.date_picker_directive.js

```diff
@@ -72,6 +83,29 @@ angular.module('lumx.date-picker', [])
             generateCalendar();
         };
 
+        // selection of year
+        $scope.selectYear = function(year) {
+
+            // update selectedDate object
+            $scope.selectedDate.date.year(year);
+            // update model
+            $scope.model = $scope.selectedDate.date.toDate();
+            // update month based on current year
+            var year_diff = year - $scope.currentSelectedYear;
+            if(year_diff < 0) {
+                $scope.month = $scope.month.subtract(12 * Math.abs(year_diff), 'month');
+            } else if(year_diff > 0) {
+                $scope.month = $scope.month.add(12 * year_diff, 'month');
+            } else {}
+            $scope.currentSelectedYear = year;
+
+            // hide year selection window
+            $scope.isYearSelected = false;
```

Comment:
Using `displayYear(false)` would be better as with this we are sure that all what the function do is done, not just changing the boolean.